clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -812,6 +814,12 @@ invariant_checker_t::check_schedule_data(per_shard_t *global)
 {
     if (serial_schedule_file_ == nullptr && cpu_schedule_file_ == nullptr)
         return;
+    // We can't verify for a partial trace (this happens when running the checker
+    // with -skip_instrs, often in -test_mode combined with another tool).
```

Comment:
This won't WAI if we skipped instrs in the middle of execution.

Reply:
You're referring to the fact that `skipped_instrs_` currently doesn't detect a skip in the middle: that looks to be true.  It has a `XXX` comment about it.  Added another such comment here.  The good news is we'll find out if someone does skip in the middle b/c the schedule invariants will fire so it's not like this lack will be some hidden problem.